### ISSUE TITLE
Regression in 3.2: Invalid charater

### ISSUE THREAD

**Issue by davidkarlsen** [flyway/flyway#971]
[ERROR] SQL State  : 22019
[ERROR] Error Code : 911
[ERROR] Message    : ORA-00911: invalid character
[ERROR] Location   : ddl/db/migration/V2__initial_schema.sql (xxxx)
[ERROR] Line       : 1
[ERROR] Statement  : CREATE TABLE FND_ACCOUNTS (

I've configured
<encoding>${project.build.sourceEncoding}</encoding> to UTF-8 - and the file is UTF-8.

It runs fine with flyway 3.1 - same drivers etc.

**Comment by axelfontaine** [comment]
Thanks for the report. Please post the full offending statement, so we can include it in our test cases.

Cheers
Axel

**Comment by sho81** [comment]
I can canfirm an Oracle 00911 bug. Take: "COMMENT ON COLUMN SATZ_MARKE.KURZZEICHEN    IS 'Kurzzeichen';" <-- Important that it is a single word comment
Then: org.flywaydb.core.internal.dbsupport.SqlStatementBuilder.addLine(String)
does not detect the "terminated" property because org.flywaydb.core.internal.dbsupport.SqlStatementBuilder.endWithOpenMultilineStringLiteral() says true and therefore terminated is NOT set to true although it should. Flyway then continues thinking that three statements are one.
line=COMMENT ON COLUMN SATZ_MARKE.KURZZEICHEN IS 'KURZZEICHEN';
delimitingTokens=[OTHER, OTHER, OTHER, OTHER, OTHER, QUOTE]
Bug seems to me in:
org.flywaydb.core.internal.dbsupport.oracle.OracleSqlStatementBuilder.cleanToken(String)
(I expect QUOTE twice)

Script part:
COMMENT ON COLUMN SATZ_MARKE.KONZERN_ID     IS 'Referenz auf einen Konzern';
COMMENT ON COLUMN SATZ_MARKE.KURZZEICHEN    IS 'Kurzzeichen';
COMMENT ON COLUMN SATZ_MARKE.STRUKTUR_ID    IS 'Referenz zur Struktur';

CREATE OR REPLACE TRIGGER SATZ_MARKE_TR1 BEFORE DELETE OR INSERT OR UPDATE ON SATZ_MARKE
...

Finally: org.flywaydb.core.internal.dbsupport.FlywaySqlScriptException.FlywaySqlScriptException(Resource, SqlStatement, SQLException)
sql="COMMENT ON COLUMN SATZ_MARKE.KURZZEICHEN    IS 'Kurzzeichen';
COMMENT ON COLUMN SATZ_MARKE.STRUKTUR_ID    IS 'Referenz zur Struktur';

CREATE OR REPLACE TRIGGER SATZ_MARKE_TR1 BEFORE DELETE OR INSERT OR UPDATE ON SATZ_MARKE
..."

**Comment by axelfontaine** [comment]
@sho81 Thanks for the analysis! This will be hugely helpful in fixing this.

**Comment by cteig** [comment]
Hi, 
I also got this Oracle 00911 error and I agree in the analysis of @sho81.  It seems like the reason that "org.flywaydb.core.internal.dbsupport.SqlStatementBuilder.endWithOpenMultilineStringLiteral() says true and therefore terminated is NOT set to true although it should" is because of the regexp-string in cleanToken-method. 

I got this error in an insert with an single quoted strings that ended with an N. When I debug the cleanToken-method I see 'myStringEndingWithA_N'-token match the regexp-string and return just a single-quote.  

I wonder if this pull request from @soemeier is the same error: https://github.com/flyway/flyway/pull/973 
The code-change in the pull-request, fix my "Illegal-character"-problem. It seems like the Oracle error was introduced iin  699571a6bc15

Cheers 
Christine Teig
